src/xmldoc.cpp

```diff
@@ -43,11 +44,8 @@ namespace libcellml {
  */
 void structuredErrorCallback(void *userData, xmlErrorPtr error)
 {
-    std::string errorString = std::string(error->message);
     // Swap libxml2 carriage return for a period.
-    if (errorString.substr(errorString.length() - 1) == "\n") {
-        errorString.replace(errorString.end() - 1, errorString.end(), ".");
-    }
+    std::string errorString = std::regex_replace(error->message, std::regex("\\n"), ".");
```

Comment:
If I recall correctly, to created a regular expression is somewhat time consuming, so might want to make `[std](std::regex("\\n"))` static.

Reply:
[bd8b62e](https://github.com/cellml/libcellml/pull/1112/commits/bd8b62e380653253b380c4e941d7cbbc597a7ac3)